sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment_async.py

```diff
@@ -589,3 +590,92 @@ def callback(pipeline_response, deserialized, _):
+                self.assertEqual(6, aspect.length)
+
+                sleek_opinion = aspect.opinions[0]
+                self.assertEqual('sleek', sleek_opinion.text)
+                self.assertEqual('positive', sleek_opinion.sentiment)
+                self.assertEqual(1.0, sleek_opinion.confidence_scores.positive)
+                self.assertEqual(0.0, sleek_opinion.confidence_scores.neutral)
+                self.assertEqual(0.0, sleek_opinion.confidence_scores.negative)
+                self.assertEqual(9, sleek_opinion.offset)
+                self.assertEqual(5, sleek_opinion.length)
+                self.assertFalse(sleek_opinion.is_negated)
+
+                premium_opinion = aspect.opinions[1]
+                self.assertEqual('premium', premium_opinion.text)
+                self.assertEqual('positive', premium_opinion.sentiment)
+                self.assertEqual(1.0, premium_opinion.confidence_scores.positive)
+                self.assertEqual(0.0, premium_opinion.confidence_scores.neutral)
+                self.assertEqual(0.0, premium_opinion.confidence_scores.negative)
+                self.assertEqual(15, premium_opinion.offset)
+                self.assertEqual(7, premium_opinion.length)
+                self.assertFalse(premium_opinion.is_negated)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @TextAnalyticsClientPreparer(client_kwargs={"api_version": ApiVersion.V3_1_PREVIEW_1})
+    async def test_aspect_based_sentiment_analysis_negated_opinion(self, client):
+        documents = [
+            "The food and service is not good"
+        ]
+
+        document = (await client.analyze_sentiment(documents=documents, show_aspects=True))[0]
+
+        for sentence in document.sentences:
+            food_aspect = sentence.aspects[0]
+            service_aspect = sentence.aspects[1]
+
+            self.assertEqual('food', food_aspect.text)
+            self.assertEqual('negative', food_aspect.sentiment)
+            self.assertEqual(0.01, food_aspect.confidence_scores.positive)
+            self.assertEqual(0.0, food_aspect.confidence_scores.neutral)
+            self.assertEqual(0.99, food_aspect.confidence_scores.negative)
+            self.assertEqual(4, food_aspect.offset)
+            self.assertEqual(4, food_aspect.length)
+
+            self.assertEqual('service', service_aspect.text)
+            self.assertEqual('negative', service_aspect.sentiment)
+            self.assertEqual(0.01, service_aspect.confidence_scores.positive)
+            self.assertEqual(0.0, service_aspect.confidence_scores.neutral)
+            self.assertEqual(0.99, service_aspect.confidence_scores.negative)
+            self.assertEqual(13, service_aspect.offset)
+            self.assertEqual(7, service_aspect.length)
+
+            food_opinion = food_aspect.opinions[0]
+            service_opinion = service_aspect.opinions[0]
+            self.assertOpinionsEqual(food_opinion, service_opinion)
```

Comment:
why compare these two opinions? are they supposed to be the same? ignore if stupid question

Reply:
In this case they should be the same, since in the sentence "The food and service is not good", the negated opinion "good" applies to both "food" and "service" aspects